Lib/shutil.py

```diff
@@ -646,6 +668,12 @@ def onerror(*args):
     elif onerror is None:
         def onerror(*args):
             raise
+    try:
+        if not os.path.exists(path):
+            raise FileNotFoundError("Cannot call rmtree on a non-existent path")
+    except:
```

Comment:
```suggestion
    except FileNotFoundError:
```

But I am not sure that this change is needed. Shouldnot `rmtree()` fail for non-existing root dir?

Reply:
This come down to which behavior is desired:

- Existing:  like the the 'rm' shell command, return an error if the supplied target is missing
- New:  the file is already gone, so who cares - we were going to delete it anyways

I went with preserving the existing behavior because there are existing test cases that expect it.  I am OK with changing the behavior, but leaving it as-is makes this a drop-in change that shouldn't affect any existing code.

Reply:
Agree, it is better to keep the existing behavior in this case.